src/Workspaces/Core/Portable/Workspace/EventMap.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis;
+
+internal sealed class EventMap
```

Comment:
is this for general purposes?  or really only for the workspace?  i'd prefer WorkspaceEventMap if hte latter.

Reply:
AFAIK, it's for workspace purposes. Renamed in next commit.